### PR TITLE
Keyinput cleanup and very basic UI

### PR DESCRIPTION
**Pull request by winnie334**
I was adding some very basic UI, when I noticed that the timescale didn't really increase as I expected. When looking around in the keyinput code I found some possible improvements and ended up throwing a large part of the code away. It still works perfectly though (actually even better!). 

Also, I noticed some strange things. When looping through the creatures in the main.js file, there are variables being created for each creature. They are then updated with some formulas, and so on, only to be thrown away at the end of the loop. Is there something I'm missing? It looks to me this is unnecessary work and possibly even a bug (if the values were intended to be saved).

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-		if (timescale >= 1) {
-			for (let ts = 0; ts < timescale; ts++) {
-				update();
-			}
-		} else {
-			tc++;
-			if (tc > 1 / timescale) {
-				update();
+	if (creatures.length > 2000) return;
+	if (timescale >= 1) {	// Can timescale ever go below 1?
```
Comment (UnknownEvil) [inline scripts/main.js]: Timescale will work if below 1, there used to be a key for it

File: scripts/main.js [hunk 2]
```diff
@@ -81,10 +81,9 @@ function update() {
 		let pos = creature.getPosition();
 
 		let color = creature.color.replace(" ", "").replace("hsl", "").replace("(", "").replace(")", "").split(",");
-
 		let lastContactX = 0;
 		let lastContactY = 0;
-		let lastContactPos = [0, 0];
+		let lastContactPos = [0, 0];	// This entire block of variables gets assigned values but they are never used?
```
Comment (UnknownEvil) [inline scripts/main.js]: Used to be used — I like to have them so I can use them later if I need to

File: scripts/main.js [hunk 3]
```diff
@@ -109,7 +108,7 @@ function update() {
 
 		let tileFood = map[pos[0]][pos[1]].food / maxTileFood;
 		let age = (creature.age / (1000 / agingSpeed));
-		let reproduceTime = creature.reproduceTime / (minReproduceTime * 2.5);
+		let reproduceTime = creature.reproduceTime / (minReproduceTime * 2.5);	// These 3 variables are assigned but never used?
```
Comment (UnknownEvil) [inline scripts/main.js]: Some of these could be useful but I'm not quite sure yet that's why they're defined but not used.

File: scripts/controls.js [hunk 4]
```diff
@@ -1,19 +1,14 @@
-function keyEvents() {
-	if (keyDown(controls.play)) {
+function checkKey(key) {
+	// checks an incoming key
+	if (key == keys[controls.play]) {
```
Comment (UnknownEvil) [inline scripts/controls.js]: I think keyDown(controls.play) looks a little cleaner than this, but that's alright

File: scripts/main.js [hunk 5]
```diff
@@ -49,11 +46,14 @@ function update() {
...
-	if (season >= growSeasonLength + dieSeasonLength || season < 0) seasonUp = !seasonUp;
+	if (season >= growSeasonLength + dieSeasonLength || season < 0) {
+		seasonUp = !seasonUp;
+		if (seasonUp) year++;
+	}
 
-	for (let i in map) {
-		for (let j in map[i]) {
-			if (map[i][j].type == 1) {
+	for (let i in map) {		// This has a big impact on performance. Wouldn't it be possible to do something alike to carykh?
```
Comment (UnknownEvil) [inline scripts/main.js]: I think we should only update inactive tiles every 20 frames or so. That'll be easy to implement and it'll cut processing time in this section to 1/20th

File: scripts/controls.js (hunk 4 above)
Comment (winnie334) [reply scripts/controls.js]: Ah sorry, you definitely have a point there! It's something I tried out at the start of my fiddling and completely forgot about. Feel free to add it back in if you want!

**Comment by winnie334** [comment]
Thanks, I'm happy you took your time to comment specifically on each of my comments as well! It clears some things up.

**Pull request merged**